tests/pep492/test_async_await.py

```diff
@@ -275,7 +278,10 @@
     result = []
     async with aiopg.sa.create_engine(loop=loop, **pg_params) as engine:
         async with engine.acquire() as conn:
-            async for value in conn.execute(sql):
+            rows = conn.execute(sql)
+            if PY_37:
```

Comment:
This is a bad decision. We are here solving the consequences and not the cause. you need to rethink it, you probably need to do this in conn.execute

Reply:
Can anyone else refactor this hack so we can move forward? I don't know what you mean by "doing this in `conn.execute`", @vir-mir.

Reply:
This is a fun one. The examples and documentation show that `aiopg.sa.SAConnection.execute` is both `await`-able and can be asynchronously iterated BUT the current implementation should only work by awaiting result before iterating. It just happens to be working without awaiting the result in CPython < 3.7 because it thinks we were supplying an old-style async iterable and that it should await the iterable first like it did pre-3.5.2.

Now that CPython doesn't revert to awaiting when __aiter__ returns an awaitable, we have a couple choices:
* Change the documentation, examples, tests to emphasize that execute is not async-for-able but its result is, like `async for row in await conn.execute(tbl.select())`

or

* Leave docs, tests, examples as-is, but make our `_SAConnectionContextManager`s behave like they should, with their `__aiter__ `providing something other than a coroutine. This could be the `_SAConnectionContextManager` instance itself, with some `__anext__` implemented, or some other wrapper around the coroutine the manager encapsulates.
  * This might be the best option given folks have probably already been using `for row in conn.execute()` in their code.

Reply:
@smagafurov I've created a possible fix PR to your PR in smagafurov/aiopg#1.